File: 2021-09/나누어떨어지는숫자배열.js
function solution(arr, divisor) {
    const answer = [];
    arr.forEach(num => {
       if(num % divisor === 0) answer.push(+num); 
    });
    return answer.length === 0 ? [-1] : answer.sort((a, b) => a-b);
}

console.log(solution([5, 9, 7, 10], 5)); // [5, 10]

/* 다른사람 풀이 배운 코드 적용 */

function solution2(arr, divisor) {
    var answer = arr.filter(v => v%divisor == 0);
    return answer.length == 0 ? [-1] : answer.sort((a,b) => a-b);
}

File: programmers/src/main/java/level1/나누어떨어지는숫자배열.java
package level1;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class 나누어떨어지는숫자배열 {

	public int[] solution(int[] arr, int divisor) {
		List<Integer> nums = new ArrayList<>();
		for (int num : arr) {
			if (num % divisor == 0) {
				nums.add(num);
			}
		}
		if (nums.isEmpty()) {
			return new int[]{-1};
		}

		Collections.sort(nums);
		int[] answer = new int[nums.size()];
		for (int i = 0; i < nums.size(); i++) {
			answer[i] = nums.get(i);
		}
		return answer;
	}

	/**
	 * filter 를 계속 까먹는다. 계속 상기시키자.
	 */
	public int[] solution2(int[] arr, int divisor) {
		int[] answer = Arrays.stream(arr).filter(factor -> factor % divisor == 0).toArray();
		if (answer.length == 0) {
			answer = new int[]{-1};
		}
		Arrays.sort(answer);
		return answer;
	}

}
